common/decorators/__init__.py

```diff
@@ -16,3 +17,15 @@ def wrapper(*args, **kwargs):
            
         return np.squeeze(func(*args, **kwargs))
     return wrapper
+
+
+def with_igrins_cal(func):
+    def add_calibration(self):
+        if ('IGRINS' in [res.id for res in self.required_resources()] and self.partner_used() > 0):
```

Comment:
I'd prefer us to not pull out resource IDs directly and compare strings. We should avoid comparing strings like this whenever possible.

In `GeminiProperties`, we have an `Instruments` `Enum` that contains an entry for `IGRINS`. Instead, this should be:

```
if GeminiProperties.Instruments.IGRINS in self.required_resources() and self.partner_used() > 0:
```

Also, no parentheses needed.